client/flutter/lib/components/forms.dart

```diff
@@ -0,0 +1,199 @@
+            items[idx].selected = false;
+          } else {
+            for (var i = 0; i < items.length; i++) {
+              items[i].selected = i == idx;
+            }
+          }
+        });
+        if (onPressed != null) {
+          onPressed(idx, items);
+        }
+      },
+    );
+  }
+}
+
+class PickOneOrMoreOptionGroup extends StatefulWidget {
+  final OptionGroupCallback onPressed;
+  final List<MultiSelectOptionItem> items;
+
+  const PickOneOrMoreOptionGroup({
+    @required this.onPressed,
+    @required this.items,
+  });
+
+  @override
+  _PickOneOrMoreOptionGroupState createState() =>
+      _PickOneOrMoreOptionGroupState(items: items, onPressed: onPressed);
+}
+
+class _PickOneOrMoreOptionGroupState extends State<PickOneOrMoreOptionGroup> {
+  final OptionGroupCallback onPressed;
+  final List<MultiSelectOptionItem> items;
+
+  _PickOneOrMoreOptionGroupState({
+    @required this.onPressed,
+    @required this.items,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return OptionGroup(
+      items: items,
+      onPressed: (idx) {
+        setState(() {
+          items[idx].selected = !items[idx].selected;
+          if (items[idx].selected) {
+            if (items[idx].noneOfTheAbove) {
+              for (var i = 0; i < idx; i++) {
+                items[i].selected = false;
+              }
+            } else {
+              for (var i = idx + 1; i < items.length; i++) {
+                if (items[i].noneOfTheAbove) {
+                  items[i].selected = false;
+                }
+              }
+            }
+          }
+        });
+        if (onPressed != null) {
+          onPressed(idx, items);
+        }
+      },
+    );
+  }
+}
+
+class OptionGroup<T extends OptionItem> extends StatelessWidget {
+  final void Function(int index) onPressed;
+  final List<T> items;
+
+  const OptionGroup({
+    @required this.onPressed,
+    @required this.items,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+        children: items.asMap().entries.map((entry) {
+      return ToggleOption(
+        title: entry.value.title,
+        color: CupertinoColors.white,
+        value: entry.value.selected,
+        onPressed: () {
+          onPressed(entry.key);
+        },
+      );
+    }).toList());
+  }
+}
+
+class ToggleOption extends StatelessWidget {
+  const ToggleOption({
+    @required this.title,
+    @required this.color,
+    this.value,
+    this.onPressed,
+  });
+
+  final String title;
+  final Color color;
+  final bool value;
+  final VoidCallback onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(
+        top: 12,
+      ),
+      child: FlatButton(
+        shape: RoundedRectangleBorder(
+          side: BorderSide(
+              width: 1.5,
+              color: value
+                  ? Constants.whoBackgroundBlueColor
+                  : Color.fromARGB(
```

Comment:
Do we want to extract a constant for this?

Reply:
Will do.